pyvista/utilities/misc.py

```diff
@@ -139,18 +139,6 @@ def copy_vtk_array(array, deep=True):
     return new_array
 
 
-def can_create_mpl_figure():  # pragma: no cover
-    """Return if a figure can be created with matplotlib."""
-    try:
-        import matplotlib.pyplot as plt
-
-        figure = plt.figure()
-        plt.close(figure)
-        return True
-    except:
-        return False
```

Comment:
I have a sneaking suspicion that there can be weird cases where this does more than "is MPL installed?". After all it checks for a window being opened and the name and docstring explicitly mention the figure.

I'm thinking that there might be issues on headless setups where matplotlib might be installed, but GUI windows can't be opened. Do you know if this can happen?

Reply:
Thank you for pointing this out. I thought there is no such case but I should be more careful. I will investigate the Matplotlib's documentation.

Reply:
To be clear I have no idea, but the test seems suspiciously specific :) Thanks for looking into this.

Reply:
I've thought about it again. If we install matplotlib and get this error I think matplotlib should give the appropriate error. If we find such a use case, let's report it to the matplotlib project.

Reply:
@tkoyama010 it might not be that simple, because there are non-interactive backends one can request, e.g. a pdf backend. It might be a legitimate use case to run that on a headless computer.

But this is only for our test suite. So if it works without this helper and corresponding skips, it's probably all good :)

Reply:
Ah, actually this is in the main library...